toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -237,14 +238,23 @@ func cleanUp(ic *ImageCustomizerParameters) error {
 func CustomizeImage(ctx context.Context, buildDir string, baseConfigPath string, config *imagecustomizerapi.Config, inputImageFile string,
 	rpmsSources []string, outputImageFile string, outputImageFormat string,
 	useBaseImageRpmRepos bool, packageSnapshotTime string,
-) error {
+) (err error) {
 	ctx, span := otel.GetTracerProvider().Tracer(OtelTracerName).Start(ctx, "customize_image")
 	span.SetAttributes(
 		attribute.String("output_image_format", string(outputImageFormat)),
 	)
-	defer span.End()
+	defer func() {
+		if err != nil {
+			span.RecordError(err)
```

Comment:
Why are you passing the error message to the telemetry in 3 different ways?

Reply:
Not passing error message anymore, but those two functions do require a string so just keeping it generic for now. These would be replaced with the error type once we have structured errors.